app/grandchallenge/components/forms.py

```diff
@@ -79,3 +86,156 @@ def save(self, *args, **kwargs):
 
     class Meta:
         fields = ("user_upload", "creator", "comment")
+
+
+class MultipleCIVCreateForm(Form):
+    _possible_widgets = {
+        *InterfaceFormField._possible_widgets,
+    }
+
+    def __init__(self, *args, instance, base_obj, user, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.instance = instance
+        self.user = user
+        self.base_obj = base_obj
+
+        # add fields for all interfaces that already exist on
+        # other display sets / archive items
+        for slug, values in base_obj.values_for_interfaces.items():
+            current_value = None
+
+            if instance:
+                current_value = instance.values.filter(
+                    interface__slug=slug
+                ).first()
+
+            self.init_interface_field(
+                interface_slug=slug, current_value=current_value, values=values
+            )
+        # Add fields for dynamically added new interfaces:
+        # These are sent along as form data like all other fields, so we can't
+        # tell them apart from the form fields initialized above. Hence
+        # the check if they already have a corresponding field on the form or not.
```

Comment:
Maybe flu-brain still but I cannot work out this construction. Wouldn't it make more sense to have two separate forms, one for adding new fields (`CreateView`) and one for updating the existing ones (`UpdateView`)?

Reply:
The form names are probably a bit confusing, let me explain, and then we can settle on better names: 
- `MultipleCIVCreateForm`: this form is the underlying form for both the DisplaySetCreate and DisplaySetUpdate form. It's a form that is dynamically constructed to contain fields for all interfaces that exist on other display sets of a given reader study, as well as potentially added new interfaces (through htmx).
- `ComponentInterfaceCreateForm`: this form is used to add one single new interface. Initially it contains an autocomplete field that lets the user choose an interface, when an interface is sent along with the get request to the view, it also contains the appropriate field for the chosen interface.